crawler/file_processing.py

```diff
@@ -638,6 +646,38 @@ def update_doc_from_source_plate(
 
         return updated_docs
 
+    def docs_to_insert_updated_with_cog_uk_ids(self, docs_to_insert: List[ModifiedRow]) -> List[ModifiedRow]:
+        """Updates sample records with COG UK IDs.
+
+        Arguments:
+            docs_to_insert {List[ModifiedRow]} -- the sample records to update.
+
+        Returns:
+            List[ModifiedRow] -- the updated samples.
+        """
+        logger.debug("Attempting to update docs with COG UK IDs")
+
+        updated_docs: List[ModifiedRow] = []
+
+        try:
+            prefix = self.centre_config[CENTRE_KEY_PREFIX]
+            count = len(docs_to_insert)
+            cog_uk_ids = generate_baracoda_barcodes(self.config, prefix, count)
+
+            for i, row in enumerate(docs_to_insert):
+                row[FIELD_MONGO_COG_UK_ID] = cog_uk_ids[i]
```

Comment:
I'm a bit ( a lot ) out of date with the crawler. Are we storing the COG UK ID in MongoDB as well as lighthouse_sample now? But your backpopulation story only overs lighthouse_sample?

Reply:
We are putting COG UK ID in Mongo for the samples coming in via the RabbitMQ route, yes.  I thought it would make sense for the same to happen here for SFTP plates/samples since the COG UK ID is assigned on creation.